src/Omnipay/Stripe/Subscription/Message/SubscriptionRequest.php

```diff
@@ -0,0 +1,158 @@
+use Stripe\Product;
+use Stripe\StripeClient;
+
+class SubscriptionRequest extends AbstractRequest
+{
+    private array $data;
+    private StripeClient $stripe;
+
+    public function __construct(array $data)
+    {
+        $this->data = $data;
+        $this->stripe = new StripeClient(Config::get('payments.stripe.secretKey'));
+    }
+
+    public function getData()
+    {
+        return $this->data;
+    }
+
+    /**
+     * @param array{user: User, invest: Invest} $data
+     */
+    public function sendData($data)
+    {
+        $user = $data['user'];
+        $invest = $data['invest'];
+
+        $project = $invest->getProject();
+
+        $price = $this->stripe->prices->create([
+            'unit_amount' => ($invest->amount + $invest->donate_amount) * 100,
+            'currency' => 'eur',
+            'recurring' => ['interval' => 'month'],
+            'product' => $this->getStripeProduct($invest)->id
+        ]);
+
+        $session = $this->stripe->checkout->sessions->create([
+            'customer' => $this->getStripeCustomer($data['user'])->id,
+            'success_url' => sprintf('%s?session_id={CHECKOUT_SESSION_ID}', $this->getRedirectUrl(
+                'invest',
+                $project->id,
+                $invest->id,
+                'complete'
+            )),
+            'cancel_url' => $this->getRedirectUrl(
+                'project',
+                $project->id
+            ),
+            'mode' => 'subscription',
+            'line_items' => [
+                [
+                    'price' => $price->id,
+                    'quantity' => 1
+                ]
+            ],
+            'metadata' => [
+                'project' => $project->id,
+                'reward' => $this->getInvestReward($invest, ''),
+                'user' => $user->id,
+            ]
+        ]);
+
+        return new SubscriptionResponse($this, $session->id);
+    }
+
+    public function completePurchase(array $options = [])
+    {
+        $session = $this->stripe->checkout->sessions->retrieve($_REQUEST['session_id']);
+
+        $this->stripe->subscriptions->update(
+            $session->subscription, [
+            'metadata' => $session->metadata->toArray()
+        ]);
+
+        return new SubscriptionResponse($this, $session->id);
+    }
+
+    private function getRedirectUrl(...$args): string
+    {
+        return sprintf(
+            '%s://%s/%s',
+            isset($_SERVER['HTTPS']) ? 'https' : 'http',
+            $_SERVER['HTTP_HOST'],
+            implode('/', $args)
+        );
+    }
+
+    /**
+     * Get the ID of the reward for the invest, or a given string in case of no reward
+     * @param Invest $invest
+     * @param string $noReward The string to return in case of no reward selected
+     * @return string
+     */
+    private function getInvestReward(Invest $invest, string $noReward): string
+    {
+        return !empty($invest->getRewards())
+            ? $invest->getRewards()[0]
```

Comment:
This line is broken. It returns an object. Since the signature of the function expects a string, this should return

`$invest->getFirstReward()->id`

Reply:
Done!